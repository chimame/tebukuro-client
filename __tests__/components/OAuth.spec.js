import React from 'react'
import {Map} from 'immutable'
import OAuthComponent from '../../client/components/OAuth'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

describe('OAuth', () => {
  const subject = (auth) => (shallow(<OAuthComponent auth={auth} />))

  test('render Signin', () => {
    const auth = Map({user: Map({isSignedIn: true, attributes: Map({image: 'example.png'})})})
    expect(toJson(subject(auth))).toMatchSnapshot()
  })

  test('render Signout', () => {
    const auth = Map({user: Map({isSignedIn: false})})
    expect(toJson(subject(auth))).toMatchSnapshot()
  })
})
