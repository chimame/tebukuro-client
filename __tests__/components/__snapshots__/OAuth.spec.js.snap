exports[`OAuth render Signin 1`] = `
<div>
  <Avatar
    src="example.png" />
  <Connect(t) />
</div>
`;

exports[`OAuth render Signout 1`] = `
<div>
  <Avatar />
  <Connect(t)
    provider="github" />
</div>
`;
